### PR TITLE
added refresh to useEntity hook

### PR DESCRIPTION
**Pull request by yousifalraheem**
## Hey, I just made a Pull Request!

This PR adds a `retry` callback to be returned in `useEntity` hook. The reason behind it is because we have an entity edit page and once changes have been submitted to the backend, there is no clean way of re-syncing the changes across all the UI elements that is using `useEntity`. Ideally, a `retry` callback is returned so that we can trigger it once changes have been successfully made. Since `useEntity` grabs the entity from the context, all components dependent on that context will receive the updates.

#### Example

```ts
const { entity, retry } = useEntity();

const updateEntity = () => {
  fetch(
    catalogApiUrl,
    { method: 'POST', body: JSON.stringify(newEntity) }
  )
    .then(retry)
    .catch(console.error);
}
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/early-socks-beg.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-catalog': minor
```
Comment (benjdlambert) [inline .changeset/early-socks-beg.md:2]: These are `patch` changes.

File: plugins/catalog-react/src/hooks/useEntity.ts [hunk 2]
```diff
@@ -51,13 +53,13 @@ export const useEntityFromUrl = (): EntityLoadingStatus => {
     }
   }, [errorApi, navigate, error, loading, entity, name]);
 
-  return { entity, loading, error };
+  return { entity, loading, error, retry };
```
Comment (benjdlambert) [inline plugins/catalog-react/src/hooks/useEntity.ts:56]: `retry` is probably not the right verb we want to use here throughout context. Maybe `refresh` is better? `retry` kind of assumes that the previous call failed.

File: .changeset/early-socks-beg.md (hunk 1 above)
Comment (yousifalraheem) [reply .changeset/early-socks-beg.md]: done

File: plugins/catalog-react/src/hooks/useEntity.ts (hunk 2 above)
Comment (yousifalraheem) [reply plugins/catalog-react/src/hooks/useEntity.ts]: done

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**